build.py

```diff
@@ -935,13 +935,11 @@ def create_dockerfile_buildbase(ddir, dockerfile_name, argmap):
     mv /tmp/boost_1_80_0/boost /usr/include/boost
 
 # Server build requires recent version of CMake (FetchContent required)
-RUN wget -O - https://apt.kitware.com/keys/kitware-archive-latest.asc 2>/dev/null | \
-      gpg --dearmor - |  \
-      tee /etc/apt/trusted.gpg.d/kitware.gpg >/dev/null && \
-    apt-add-repository 'deb https://apt.kitware.com/ubuntu/ focal main' && \
-    apt-get update && \
-    apt-get install -y --no-install-recommends \
-      cmake-data=3.21.1-0kitware1ubuntu20.04.1 cmake=3.21.1-0kitware1ubuntu20.04.1
+ARG CMAKE_VERSION=3.25.2
+RUN apt update && apt install -y curl tar && \
+    curl -OL https://github.com/Kitware/CMake/releases/download/v${CMAKE_VERSION}/cmake-${CMAKE_VERSION}-linux-$(uname -m).tar.gz && \
+    tar -xvf cmake-${CMAKE_VERSION}-linux-$(uname -m).tar.gz -C /usr/local  --strip 1 --exclude=doc --exclude=man && \
+    rm -rf cmake-${CMAKE_VERSION}-linux-$(uname -m).tar.gz
```

Comment:
Why are we changing the cmake installation? I think the previous approach was cleaner since it used apt.

Reply:
ONNXRUNTIME set a minimum required version 3.24 without such a change we can't build the `onnxruntime_backend` for ORT 1.14.1 . 
CMake is backward compatible and we shouldn't have issue with it.

Reply:
Is that version not included in the `apt.kitware.com/ubuntu`?

Reply:
I wasn't able to install it using kitware repo as of time i added it.

Reply:
I just tried it seems like `3.24.0` is available now:

```
apt-get install -y --no-install-recommends cmake-data=3.24.0-0kitware1ubuntu20.04.1 cmake=3.24.0-0kitware1ubuntu20.04.1
Reading package lists... Done
Building dependency tree
Reading state information... Done
Suggested packages:
  cmake-doc ninja-build
The following NEW packages will be installed:
  cmake cmake-data
0 upgraded, 2 newly installed, 0 to remove and 73 not upgraded.
Need to get 13.5 MB of archives.
After this operation, 46.7 MB of additional disk space will be used.
Get:1 https://apt.kitware.com/ubuntu focal/main amd64 cmake-data all 3.24.0-0kitware1ubuntu20.04.1 [1986 kB]
Get:2 https://apt.kitware.com/ubuntu focal/main amd64 cmake amd64 3.24.0-0kitware1ubuntu20.04.1 [11.6 MB]
Fetched 13.5 MB in 1s (9295 kB/s)
Selecting previously unselected package cmake-data.
(Reading database ... 25128 files and directories currently installed.)
Preparing to unpack .../cmake-data_3.24.0-0kitware1ubuntu20.04.1_all.deb ...
Unpacking cmake-data (3.24.0-0kitware1ubuntu20.04.1) ...
Selecting previously unselected package cmake.
Preparing to unpack .../cmake_3.24.0-0kitware1ubuntu20.04.1_amd64.deb ...
Unpacking cmake (3.24.0-0kitware1ubuntu20.04.1) ...
Setting up cmake-data (3.24.0-0kitware1ubuntu20.04.1) ...
Setting up cmake (3.24.0-0kitware1ubuntu20.04.1) ...
root@6978db0dc005:/opt/tritonserver# cmake --version
cmake version 3.24.0

CMake suite maintained and supported by Kitware (kitware.com/cmake).
```

Reply:
Ok I will change it back